lab3/client/src/reducer.js

```diff
@@ -0,0 +1,16 @@
+export default function reducer (state, action) {
+	switch (action.type) {
+	case 'EXAMPLE_MUTATION':
+		state.example = action.payload;
+		return state;
+		break;
+	case 'BUY_GENERATOR':
+		state.counter = action.payload;
+		state.generator = action.payload;
```

Comment:
ditto. Need to loop through a list of generators under `state.generators` to find out which generator to increase its quantity.